### PR TITLE
Fix POS Stripe Terminal nonce generation

### PR DESCRIPTION
**Pull request by kilbot**
## Summary
- Create the Stripe Terminal AJAX nonce for the POS cashier recorded on the order (`_pos_user`) instead of the temporary order/customer user used while WCPOS renders the order-pay iframe.
- Restore the original current user immediately after nonce creation.
- Guard the POS-only return URL helper so standalone checkout flows do not fatal if WooCommerce POS is unavailable.
- Add regression coverage for the POS cashier nonce path and the missing POS helper path.

## Root cause
PR #48 added `check_ajax_referer( 'stripe_terminal_nonce', 'nonce', false )` to the AJAX endpoints. In the POS checkout iframe, WCPOS renders the WooCommerce order-pay template after switching `current_user` to the order customer/guest. That meant the localized nonce was generated for the guest/customer context, but `admin-ajax.php` later validated it under the logged-in cashier session. The nonce was present in the request, but invalid for the submitting user, so `create_payment_intent` returned `Invalid request` before Stripe was called.

## Validation
- `composer test -- --filter GatewayTest` — OK (2 tests, 3 assertions)
- `composer test` — OK (220 tests, 326 assertions)
- `composer lint` — OK
- Hot-patched `https://dev-pro.wcpos.com/` and reproduced the POS flow against the real Stripe test server. Before the fix the same flow returned `{"success":false,"data":"Invalid request"}` for `stripe_terminal_create_payment_intent`. After the fix, `stripe_terminal_validate_service` and `stripe_terminal_get_readers` both succeeded, and `stripe_terminal_create_payment_intent` reached Stripe and failed with the Stripe account/currency response: `Currency GBP is not supported by Stripe Terminal. Supported currencies: USD`. That confirms the nonce/auth blocker is removed and the request now reaches Stripe.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevents fatal errors when POS helper functions are unavailable and adds clearer error responses for missing/expired security tokens.

* **Improvements**
  * Frontend payment nonce generation now respects POS cashier session context on order-pay pages to support terminal payments.

* **Tests**
  * Added unit tests covering nonce behavior, user switching, and order-received URL handling; extended AJAX tests for missing/invalid nonce cases.

* **Chores**
  * Bumped plugin version to 0.0.21 and updated changelog.

### CODE REVIEW

File: includes/Gateway.php [hunk 1]
```diff
@@ -434,7 +435,7 @@ public function enqueue_payment_scripts(): void {
 			'stripeTerminalData',
 			array(
 				'ajaxUrl'  => admin_url( 'admin-ajax.php' ),
-				'nonce'    => wp_create_nonce( 'stripe_terminal_nonce' ),
+				'nonce'    => $this->create_ajax_nonce( $order ),
```
[the remark this answers is not in the document]
Comment (kilbot) [reply includes/Gateway.php:439]: Fixed in 3ed2a04 — `wc_get_order()` results are now normalized to `WC_Abstract_Order|null` before passing the value into `create_ajax_nonce()`, so stale or malformed order-pay URLs fall back to the default nonce path instead of throwing a `TypeError`.

File: includes/Gateway.php [hunk 2]
```diff
@@ -469,6 +470,35 @@ public function enqueue_payment_scripts(): void {
...
+	 * WooCommerce POS renders order-pay templates after switching the current
+	 * WordPress user to the order customer. The subsequent admin-ajax.php request
+	 * is submitted by the logged-in POS cashier, so the nonce must be created for
+	 * that cashier session or check_ajax_referer() rejects the payment request.
+	 *
+	 * @param WC_Abstract_Order|null $order Order currently being paid.
+	 * @return string
+	 */
+	private function create_ajax_nonce( ?WC_Abstract_Order $order = null ): string {
+		$pos_user_id = $order ? absint( $order->get_meta( '_pos_user', true ) ) : 0;
```
[the remark this answers is not in the document]
Comment (kilbot) [reply includes/Gateway.php:490]: Fixed in 3ed2a04 — the `_pos_user` nonce override now only runs during `woocommerce_pos_request()`. Non-POS order-pay flows keep the default current-user nonce behavior.
